models/asymkey/gpg_key.go

```diff
@@ -92,15 +92,17 @@ func CountUserGPGKeys(ctx context.Context, userID int64) (int64, error) {
 	return db.GetEngine(ctx).Where("owner_id=? AND primary_key_id=''", userID).Count(&GPGKey{})
 }
 
-// GetGPGKeyByID returns public key by given ID.
-func GetGPGKeyByID(ctx context.Context, keyID int64) (*GPGKey, error) {
+func GetGPGKeyForUserByID(ctx context.Context, ownerID, keyID int64) (*GPGKey, error) {
 	key := new(GPGKey)
 	has, err := db.GetEngine(ctx).ID(keyID).Get(key)
 	if err != nil {
 		return nil, err
 	} else if !has {
 		return nil, ErrGPGKeyNotExist{keyID}
 	}
+	if key.OwnerID != ownerID {
+		return nil, ErrGPGKeyNotExist{keyID}
+	}
```

Comment:
Use `db.GetEngine(ctx).Where(id = keyID and owner_id=OwnerID)`, then no need to do this `if` check?